release-notes/6.0/preview/api-diff/preview6/Asp.Net/6.0-preview6_Microsoft.AspNetCore.Http.md

```diff
@@ -0,0 +1,37 @@
+# Microsoft.AspNetCore.Http
+
+``` diff
+ namespace Microsoft.AspNetCore.Http {
+     public abstract class ConnectionInfo {
++        public virtual void RequestClose();
+     }
+     public static class HttpProtocol {
++        public static readonly string Http09;
++        public static bool IsHttp09(string protocol);
+     }
+     public interface IHeaderDictionary : ICollection<KeyValuePair<string, StringValues>>, IDictionary<string, StringValues>, IEnumerable, IEnumerable<KeyValuePair<string, StringValues>> {
++        StringValues SecWebSocketExtensions { get; set; }
+     }
+     public static class RequestDelegateFactory {
+-        public static RequestDelegate Create(Delegate action);
+
++        public static RequestDelegate Create(Delegate action, IServiceProvider serviceProvider);
+-        public static RequestDelegate Create(MethodInfo methodInfo);
+
+-        public static RequestDelegate Create(MethodInfo methodInfo, Func<HttpContext, object> targetFactory);
+
++        public static RequestDelegate Create(MethodInfo methodInfo, IServiceProvider serviceProvider);
++        public static RequestDelegate Create(MethodInfo methodInfo, IServiceProvider serviceProvider, Func<HttpContext, object> targetFactory);
```

Comment:
Just as a heads up, we don't expect this to be the *final* form of RequestDelegate.Create(), but it is what we intend to ship for preview6. https://github.com/dotnet/aspnetcore/issues/33700 discusses using an options type for the IServiceProvider and other things.